pkg/stub/handler.go

```diff
@@ -163,10 +225,34 @@ func (h *Handler) syncIngressUpdate(ci *ingressv1alpha1.ClusterIngress) error {
 	return nil
 }
 
-func (h *Handler) deleteIngress(ci *ingressv1alpha1.ClusterIngress) error {
+// ensureRouterDeleted ensures that any router resources associated with the
+// clusteringress are deleted.
+func (h *Handler) ensureRouterDeleted(ci *ingressv1alpha1.ClusterIngress) error {
 	ds, err := h.manifestFactory.RouterDaemonSet(ci)
 	if err != nil {
 		return fmt.Errorf("couldn't build DaemonSet object for deletion: %v", err)
 	}
-	return sdk.Delete(ds)
+	err = sdk.Delete(ds)
+	if !errors.IsNotFound(err) {
+		return err
+	}
+	return nil
+}
+
+// RemoveString returns a newly created []string that contains all items from slice that
+// are not equal to s.
+func RemoveString(slice []string, s string) []string {
```

Comment:
removeString() if we don't expect to be used in other files.

Reply:
I wish the upstream `slice` package were in a better place so we could use it and some other nice stuff I'll have to copy in 😞